pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -285,6 +286,17 @@ public void testSetNull() throws SQLException {
     pstmt.executeUpdate();
 
     pstmt.close();
+
+    pstmt = con.prepareStatement("select 'ok' where ?=? or (? is null) ");
+    pstmt.setObject(1, UUID.randomUUID(), Types.OTHER);
+    pstmt.setNull(2, Types.OTHER, "uuid");
+    pstmt.setNull(3, Types.OTHER, "uuid");
+    ResultSet rs = pstmt.executeQuery();
+
+    while (rs.next()) {
```

Comment:
The statement should return rows, so please add an assertion (to ensure nulls are passed properly).